src/socket.c

```diff
@@ -32,16 +32,15 @@
 void socket_init(socket_ctx *ctx, char *path) {
 	printf("initialize socket\n");
 	ctx->socket.sun_family = AF_UNIX;
-	strcpy(ctx->socket.sun_path, path);
+	strncpy(ctx->socket.sun_path, path, strlen(path));
```

Comment:
For reference: In fastd, I use a variable-size sockaddr_un to allow arbitrary path lengths: https://git.universe-factory.net/fastd/tree/src/status.c#n244 (although I just noticed that my code is not completely correct either, as buf is not aligned to the minimum alignment of a sockaddr_un; I'll fix that soon...)